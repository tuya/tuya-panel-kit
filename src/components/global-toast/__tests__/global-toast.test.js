import React from 'react';
import renderer from 'react-test-renderer';
import GlobalToast from '../index';

describe('GlobalToast Component', () => {
  it('basic render', () => {
    const component = renderer
      .create(
        GlobalToast.show({
          text: '提示性文案建议最多展示十六个字符',
          showIcon: false,
          textStyle: { fontSize: 18, color: '#333' },
          contentStyle: {},
          onFinish: jest.fn(),
        })
      )
      .toJSON();
  });

  it('render width loading', () => {
    const component = renderer
      .create(
        GlobalToast.show({
          text: '设置成功',
          textStyle: { fontSize: 18, color: '#333' },
          showPosition: 'center',
          onFinish: jest.fn(),
        })
      )
      .toJSON();
  });

  it('render width update', () => {
    const component = renderer.create(
      GlobalToast.show({
        text: '设置成功',
        showPosition: 'center',
        onFinish: jest.fn(),
      })
    );
    component.update(GlobalToast.hide());
  });
});

describe('GlobalToast', () => {
  it('basic render', () => {
    const onFinish = jest.fn();
    const component = renderer.create(<GlobalToast show={true} onFinish={onFinish} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
