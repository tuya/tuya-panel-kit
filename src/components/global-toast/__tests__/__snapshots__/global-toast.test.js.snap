// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalToast basic render 1`] = `
<View
  pointerEvents="none"
  style={
    Array [
      false,
      Object {
        "alignItems": "center",
        "bottom": 0,
        "flex": 1,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
      null,
      Object {
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0,0,0,.7)",
        "borderRadius": 24,
        "justifyContent": "center",
        "marginBottom": 64,
        "marginHorizontal": 44,
        "opacity": 0,
        "paddingHorizontal": 36,
        "paddingVertical": 36,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          Object {
            "height": 84,
            "width": 84,
          },
          false,
          null,
        ]
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
            false,
            Object {
              "flex": 0,
              "height": 84,
              "width": 84,
            },
          ]
        }
        vbHeight={1024}
        vbWidth={1024}
      >
        <RNSVGPath
          d="M512 0c282.770286 0 512 229.229714 512 512s-229.229714 512-512 512S0 794.770286 0 512 229.229714 0 512 0z m0 48.761905C256.170667 48.761905 48.761905 256.170667 48.761905 512s207.408762 463.238095 463.238095 463.238095 463.238095-207.408762 463.238095-463.238095S767.829333 48.761905 512 48.761905z m209.92 324.022857a24.380952 24.380952 0 0 1 2.194286 32.182857l-2.023619 2.29181-247.076572 249.441523a24.380952 24.380952 0 0 1-32.597333 1.852953l-2.29181-2.096762-116.370285-120.978286a24.380952 24.380952 0 0 1 32.914285-35.864381l2.243048 2.072381 99.035429 102.936381 229.497904-231.69219a24.380952 24.380952 0 0 1 34.499048-0.146286z"
          fill={
            Array [
              0,
              1,
              1,
              1,
              1,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          propList={
            Array [
              "stroke",
              "strokeWidth",
              "fill",
            ]
          }
          stroke={
            Array [
              0,
              1,
              1,
              1,
              1,
            ]
          }
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
        />
      </RNSVGSvgView>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "flexWrap": "wrap",
            "fontSize": 14,
            "lineHeight": 20,
            "textAlign": "center",
          },
          Object {
            "marginTop": 12,
          },
          null,
        ]
      }
    >
      成功文案
    </Text>
  </View>
</View>
`;
